yb-voyager/cmd/exportSchema.go

```diff
@@ -309,32 +316,75 @@ func applyShardedTablesRecommendation(shardedTables []string) error {
 	return nil
 }
 
-func applyShardingRecommendation(sqlInfo sqlInfo, lastStmtSetOrSelect bool) string {
-	newSQL := strings.TrimRight(sqlInfo.formattedStmt, "; ")
-	newSQL += " WITH (COLOCATION = false);\n\n\n"
-	return prependSpacing(newSQL, lastStmtSetOrSelect)
-}
+/*
+applyShardingRecommendation uses pg_query module to parse the given SQL stmt
+In case of any errors or unexpected behaviour it return the original DDL
+so in worse only recommendation of that table won't be followed.
 
-func appendSpacing(sql string, lastStmtSetOrSelect bool) string {
-	return prependSpacing(sql+"\n\n\n", lastStmtSetOrSelect)
-}
+# It can handle cases like multiple options in WITH clause
+
+Drawback: pg_query module doesn't have functionality to format the query after parsing
+so the CREATE TABLE for sharding recommended tables will be one-liner
+*/
+func applyShardingRecommendation(sqlInfo *sqlInfo) (string, error) {
+	stmt := sqlInfo.stmt
+	parseTree, err := pg_query.Parse(stmt)
+	if err != nil {
+		return stmt, fmt.Errorf("error parsing the stmt-%s: %v", stmt, err)
+	}
 
-func prependSpacing(sql string, condition bool) string {
-	if condition {
-		return "\n\n" + sql
+	if len(parseTree.Stmts) == 0 {
+		log.Infof("parse tree is empty, returning")
+		return stmt, nil
```

Comment:
shouldn't we return a non-nil error here?

Reply:
we don't know yet, when will this be empty. maybe empty strings, some valid syntax which is not covered by the library!

Reply:
Here you can log (WARN) along with the sqlinfo.stmt?